crates/primitives/src/prune/mod.rs

```diff
@@ -91,21 +91,76 @@ impl ReceiptsLogPruneConfig {
 #[derive(Debug, PartialEq, Eq, Clone, Copy)]
 pub enum PruneProgress {
     /// There is more data to prune.
-    HasMoreData,
+    HasMoreData(PruneInterruptReason),
     /// Pruning has been finished.
     Finished,
 }
 
+/// Reason for interrupting a prune job.
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum PruneInterruptReason {
+    /// Prune job timed out.
+    Timeout,
+    /// Limit on the number of deleted entries (rows in the database) per prune job was reached.
+    LimitEntriesDeleted,
+}
+
+impl PruneInterruptReason {
+    /// Returns `true` if reason is timeout.
+    pub fn is_timeout(&self) -> bool {
+        matches!(self, Self::Timeout)
+    }
+
+    /// Returns `true` if reason is reaching limit on deleted entries.
+    pub fn is_entries_limit_reached(&self) -> bool {
+        matches!(self, Self::LimitEntriesDeleted)
+    }
+}
+
 impl PruneProgress {
-    /// Creates new [PruneProgress] from `done` boolean value.
+    /// Creates new [PruneProgress] that summarises prune job.
     ///
-    /// If `done == true`, returns [PruneProgress::Finished], otherwise [PruneProgress::HasMoreData]
-    /// is returned.
-    pub fn from_done(done: bool) -> Self {
+    /// If `done == true`, returns [`PruneProgress::Finished`], otherwise
+    /// [PruneProgress::HasMoreData] is returned.
```

Comment:
nit: consistency between `[]` and `[``]` usage
```suggestion
    /// If `done == true`, returns [PruneProgress::Finished], otherwise
    /// [PruneProgress::HasMoreData] is returned.
```